### PR TITLE
forestrie trust level updates supporting confirmation of forestrie events

### PR DESCRIPTION
**Pull request by robinbryce**
* Add the confirmation states representing the forestrie trust levels as described here: https://github.com/datatrails/epic-8120-scalable-proof-mechanisms/blob/main/event-trust-levels.md
* Add a oneof to EventResponse to take the proof_details for the merklelog proof mechanism
* Add the necessary proto messages to support forestrie notifying assetstorev2confirmer of event confirmationstatus COMMITTED
* Fix the include paths for openapiv2
* simplify the include path discovery
* add docker-all target for people who don't want to build on their host (which is a _lot_ faster)
* export the proto dependencies under proto-include so consumers of go-datatrails-common-api do not need to replicate the location and fetching of same

### CODE REVIEW

File: Taskfile.yml [hunk 1]
```diff
@@ -67,22 +67,42 @@ tasks:
   ### -------------------------
   # Primary workflow tasks 
   ### -------------------------
-  all:
-    desc: "do everything necessary after clone (or rebase)"
+  dockerall:
```
Comment (honourfish) [inline Taskfile.yml:70]: minor: I think i would prefer that the consistent environment have the name all, and the local environment would be named differently? like `local-all`

Minor as it is clear in the descriptions what these tasks do

File: datatrails-common-api/assets/v2/assets/enums.proto [hunk 2]
```diff
@@ -7,6 +7,13 @@ enum ConfirmationStatus {
     PENDING = 1;        // not yet committed
     CONFIRMED = 2;      // committed
     FAILED = 3;         // permanent failure
+
+	// Regarding the new statuses for forestrie,  See
+	// https://github.com/datatrails/epic-8120-scalable-proof-mechanisms/blob/main/event-trust-levels.md
+    STORED = 4; // forestrie, "its in the db"
+    COMMITTED = 5; // forestrie, "you can know if its changed"
+    // CONFIRMED = 6; // forestrie, "You can easily prove it changed"
+    UNEQUIVOCAL = 7; // forestrie, "You easily prove it was publicly available to all"
```
Comment (honourfish) [inline datatrails-common-api/assets/v2/assets/enums.proto:16]: I prefer UNEQUIVOCAL over ANCHORED.

File: datatrails-common-api/assets/v2/assets/enums.proto [hunk 3]
```diff
@@ -7,6 +7,13 @@ enum ConfirmationStatus {
     PENDING = 1;        // not yet committed
     CONFIRMED = 2;      // committed
     FAILED = 3;         // permanent failure
+
+	// Regarding the new statuses for forestrie,  See
+	// https://github.com/datatrails/epic-8120-scalable-proof-mechanisms/blob/main/event-trust-levels.md
+    STORED = 4; // forestrie, "its in the db"
+    COMMITTED = 5; // forestrie, "you can know if its changed"
+    // CONFIRMED = 6; // forestrie, "You can easily prove it changed"
```
Comment (wp0pw) [inline datatrails-common-api/assets/v2/assets/enums.proto:15]: why is this one commented out?

File: datatrails-common-api/assets/v2/assets/eventresponse.proto [hunk 4]
```diff
@@ -164,4 +167,17 @@ message EventResponse {
         "Identity of the tenant the that created this event"
         max_length: 1024
     }];
+
+    // An event has exactly one proof mechanism. This field caputures the proof
```
Comment (honourfish) [inline datatrails-common-api/assets/v2/assets/eventresponse.proto:171]: Minor: I think comments like this are transposed onto the swagger when it gets autogenerated. We need verbose commentary like this, but not exposed to the user via swagger

File: datatrails-common-api/assets/v2/assets/enums.proto (hunk 2 above)
Comment (robinbryce) [reply datatrails-common-api/assets/v2/assets/enums.proto:16]: So does @JAG-UK I think. It avoids confusion with simple hash too

**Review by honourfish — DISMISSED** [review]

File: datatrails-common-api/assets/v2/assets/eventresponse.proto (hunk 4 above)
Comment (robinbryce) [reply datatrails-common-api/assets/v2/assets/eventresponse.proto:171]: ah ok, is there a way to add the comment so it _doesn't_ get into the swagger ?

File: datatrails-common-api/assets/v2/assets/merklelogentry.proto [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    string tenant_identity = 1;
+    string event_identity = 2;
+    /* The time portion of idtimestamp that contributed to the hash of the event
+     * (the idtimestamp is _also_ included.
+     * This must be copied into event.timestamp_committed when the saas db is updated */
+    google.protobuf.Timestamp timestamp = 6;
+
+    uint32 log_version = 3;
+    uint32 log_epoch = 4;
+    MerkleLogCommit committed = 5;
```
[the remark this answers is not in the document]
Comment (robinbryce) [reply datatrails-common-api/assets/v2/assets/merklelogentry.proto:33]: how do you mean ? are you suggesting just a rename committed -> commit or something else ? The act of committing happens in forestrie. This field is a record of what happened their and is past tense, but tomatO tomAto. If that is all you mean ?

File: datatrails-common-api/assets/v2/assets/eventresponse.proto (hunk 4 above)
Comment (honourfish) [reply datatrails-common-api/assets/v2/assets/eventresponse.proto:171]: My memory is hazy, but i think all comments directly above properties,messages or rpc definitions all go to swagger, so maybe a blank line might work?

File: datatrails-common-api/assets/v2/assets/merklelogentry.proto (hunk 5 above)
[the remark this answers is not in the document]
Comment (wp0pw) [reply datatrails-common-api/assets/v2/assets/merklelogentry.proto:33]: yep - as commited for me would be something boolean like - whereas this is all done and dusted and just has some info about actual commit that already happened (always afaik)

File: Taskfile.yml (hunk 1 above)
Comment (robinbryce) [reply Taskfile.yml]: done

File: datatrails-common-api/assets/v2/assets/enums.proto (hunk 3 above)
Comment (robinbryce) [reply datatrails-common-api/assets/v2/assets/enums.proto]: moved the forestrie comment to the original constant and removed the rest

File: datatrails-common-api/assets/v2/assets/eventresponse.proto (hunk 4 above)
Comment (robinbryce) [reply datatrails-common-api/assets/v2/assets/eventresponse.proto:171]: Will defer this for now but remember it. LMK what the best way is to capture the more verbose comentary

File: datatrails-common-api/assets/v2/assets/merklelogentry.proto (hunk 5 above)
[the remark this answers is not in the document]
Comment (robinbryce) [reply datatrails-common-api/assets/v2/assets/merklelogentry.proto]: done

**Review by honourfish — APPROVED** [review]

File: datatrails-common-api/assets/v2/assets/merklelogentry.proto [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+    uint32 log_version = 3;
+    uint32 log_epoch = 4;
+    MerkleLogCommit commit = 5;
+}
+
+
+
+// The details stored in the SaaS db for a proof mech MERKLE_LOG commitment
+message MerklLogEntry {
```
Comment (wp0pw) [inline datatrails-common-api/assets/v2/assets/merklelogentry.proto:39]: should this be Merkl**e**LogEntry? and also can we do same with commited -> commit here?

**Pull request merged**